packages/wrangler/src/dev.tsx

```diff
@@ -476,6 +476,11 @@ async function updateDevEnvRegistry(
 	devEnv: DevEnv,
 	registry: WorkerRegistry | undefined
 ) {
+	// Make sure we're not patching an empty config
+	if (!devEnv.config.latestConfig) {
```

Comment:
What's this change fixing?

Reply:
This is just moving this check higher up the function. Previously we got bound workers and _then_ waited for a config update, but since getting bound workers depends on the config we should've been waiting here instead

Reply:
Aha! Does this explain the race we were seeing?

Reply:
One of them, yes! Another was in the bundle controller, and there might still be others lurking